### ISSUE TITLE
Selection Bug

### ISSUE THREAD

**Issue by ishdemon** [davideas/FlexibleAdapter#606]
I have been tweaking the demo app in selection modes and I found this weird bug. After you turn off the screen and turn on and if you select any other item, the selection becomes ambiguous and glitchy resulting in multiple selections. It gets cleared once you scroll past them. Please note it's not the long press for multiselection, it happens in normal press.

![Initial state](https://user-images.githubusercontent.com/13925683/39688873-54f0d034-51f2-11e8-873a-6f749f522673.png)

![After screen off and on](https://user-images.githubusercontent.com/13925683/39688845-320b40cc-51f2-11e8-8242-3f9acb366963.png)

![After Scrolling past those selected items](https://user-images.githubusercontent.com/13925683/39688888-6337889a-51f2-11e8-9176-a66837866049.png)

**Comment by ishdemon** [comment]
One Temporary workaround I have found is Refreshing the dataset in onResume(); But it's not feasible as the dataset is reloading unnecessary.

**Comment by davideas** [comment]
@ishdemon, thank you for the messages.
I already verified your steps, indeed something is not working as expected, but I don't know if the demoApp or the library, I will find out and fix it.

**Comment by ishdemon** [comment]
Thanks. Looking forward to it.

**Comment by davideas** [comment]
@ishdemon, yes, I confirm that is a small bug, introduced in 5.0.2 when clearing the cache for view bounds in `onSaveInstanceState`, but the mistake is that views are not being bound again on restore, so I should not clear the cache previously, but only when the full dataSet is updating:
1) when `updateDataSet(List)` is invoked (same as `updateDataSet(List, false)`);
2) when `updateDataSet(List, true)` is invoked with `notifyChangeOfUnfilteredItems=true (default)`.

> ⚠️ **Warning**: Once again, I remind that, when using **FlexibleAdapter** we should **never** call official methods **notify***** but call the relative methods I created, because there are others operations on going, like the one above.

> ℹ️ **I also remind** that, view holders in this cache are removed when views are recycled. The cache keeps only very few elements constantly in memory, so it's not an issue.
The cache helps to see animations on going, because we are not interested in rebinding the content but making a transition of the view from the old state to the new state, so we allow animations to be performed without interruption.
